packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -294,7 +294,7 @@ class _DevFSHttpWriter {
     } catch (e) {
       if (retry < kMaxRetries) {
         printTrace('Retrying writing "$deviceUri" to DevFS due to error: $e');
-        _scheduleWrite(deviceUri, content, progressReporter, retry + 1);
+        await _scheduleWrite(deviceUri, content, progressReporter, retry + 1);
```

Comment:
Nonetheless, this future should be ignored.  It looks like we use `_completer` below to complete a future once all pending writes are done -- and we await that future in `write()`.  So if we await here, we could deadlock.

(side-note: this code should probably be restructured to be simpler...)